src/main/java/net/imglib2/cache/util/FunctorLoader.java

```diff
@@ -0,0 +1,44 @@
+package net.imglib2.cache.util;
+
+import java.util.stream.IntStream;
+
+import net.imglib2.FinalInterval;
+import net.imglib2.Interval;
+import net.imglib2.cache.CacheLoader;
+import net.imglib2.img.cell.Cell;
+import net.imglib2.img.cell.CellGrid;
+
+public class FunctorLoader< A > implements CacheLoader< Long, Cell< A > >
+{
+
+	public interface Functor< T, R > {
```

Comment:
That's what I was using initially but `Function.apply` does not throw `Exception`. We should keep throwing `Exception` as an option for loaders like http loaders or ZeroMQ loaders to indicate failure like inability to create a connection etc.